repl/src/main/scala-2.12/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -105,6 +108,13 @@ class SparkILoop(in0: Option[BufferedReader], out: JPrintWriter)
     echo("Type :help for more information.")
   }
 
+  override def processLine(line: String): Boolean = {
+    inactivityTimeout.stopInactivityTimer()
+    val result = super.processLine(line)
```

Comment:
What happens when the line is evaluated and it throws an exception? 
Do the underlying `processLine` guarantees that exceptions are not propagated into the caller? Because if there is no such guarantees then we should wrap this into a `try {..} finally {...}` and stop the timer in the `finally`.

Reply:
Thanks for your feedback! I have now updated the PR with the suggested change.